### PR TITLE
Adds sanity check to disposal climbing

### LINKED ISSUE #10278: Climb into a destroyed disposal = nullspace
Be a simple mob that can destroy things
Start climbing into a disposal
Attack the disposal to delete it
Climb into nullspace

You probably don't have to be the one to destroy the disposal, the point is that the climbing check needs a sanity at the end of the climbing in phase to prevent this.

### PR DESCRIPTION
**Pull request by maartenberg**
Fixes #10278

### CODE REVIEW

File: code/modules/recycling/disposal.dm [hunk 1]
```diff
@@ -152,6 +152,8 @@
 		target.visible_message("<span class='danger'>[user] starts putting [target] into [src].</span>", \
 								"<span class='userdanger'>[user] starts putting you into [src]!</span>")
 	if(do_mob(user, target, 20))
+		if (!src.loc)
```
Comment (xxalpha) [inline code/modules/recycling/disposal.dm]: >src.loc

ahhhh

Comment (spasticVerbalizer) [inline code/modules/recycling/disposal.dm]: Why is that bad? Varedit shows the chute continues to exist if you break it while going in, but its loc gets set to null, so this works

Comment (Menshin) [inline code/modules/recycling/disposal.dm]: Should probably rather check for _gc_destroyed_.

Comment (phil235) [inline code/modules/recycling/disposal.dm]: src.loc is alright. The same thing is used in many other places in the code.

Comment (xxalpha) [inline code/modules/recycling/disposal.dm]: I admit src.loc is more acceptable, for readability.

**Pull request merged**